instrumentation/vertx/vertx-http-client/vertx-http-client-3.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/vertx/v3_0/client/VertxHttpClientTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.vertx.v3_0.client;
+
+import client.VertxSingleConnection;
+import io.opentelemetry.api.common.AttributeKey;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.AbstractHttpClientTest;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpClientInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpClientResult;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpClientTestOptions;
+import io.opentelemetry.instrumentation.testing.junit.http.SingleConnection;
+import io.opentelemetry.semconv.SemanticAttributes;
+import io.vertx.core.Vertx;
+import io.vertx.core.VertxOptions;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.http.HttpClientOptions;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpMethod;
+import java.net.URI;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class VertxHttpClientTest extends AbstractHttpClientTest<HttpClientRequest> {
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = HttpClientInstrumentationExtension.forAgent();
+
+  private final HttpClient httpClient = buildClient();
+
+  private static HttpClient buildClient() {
+    Vertx vertx = Vertx.vertx(new VertxOptions());
+    HttpClientOptions clientOptions =
+        new HttpClientOptions().setConnectTimeout(Math.toIntExact(CONNECTION_TIMEOUT.toMillis()));
+    return vertx.createHttpClient(clientOptions);
+  }
+
+  @Override
+  public HttpClientRequest buildRequest(String method, URI uri, Map<String, String> headers) {
+    HttpClientRequest request = httpClient.requestAbs(HttpMethod.valueOf(method), uri.toString());
+    headers.forEach(request::putHeader);
+    return request;
+  }
+
+  private static CompletableFuture<Integer> sendRequest(HttpClientRequest request) {
+    CompletableFuture<Integer> future = new CompletableFuture<>();
+    request
+        .handler(response -> future.complete(response.statusCode()))
+        .exceptionHandler(future::completeExceptionally)
+        .end();
+    return future;
+  }
+
+  @Override
+  public int sendRequest(
+      HttpClientRequest request, String method, URI uri, Map<String, String> headers)
+      throws InterruptedException {
```

Comment:
I think you could have `throws Exception` here and could skip wrapping the exceptions into runtime exception. This in turn should allow removing the `setClientSpanErrorMapper` call.

Reply:
Right. Updated.